client/components/ProductDetail/UpdateCart.jsx

```diff
@@ -1,64 +1,86 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
-import AddToCart from './AddToCart.jsx';
+import ls from 'local-storage';
+import { Button } from '../Shared/Form';
 
-const UpdateCart = (props) => {
-  var availability = [{
-    size: 'Select Size',
-    quantity: null
-  }];
+const UpdateCart = ({ style, product, addToCart }) => {
+  //Define Variables for Functional Component
+  const [selectedSize, setSelectedSize] = useState(0);
+  const [selectedQuantity, setSelectedQuantity] = useState(0);
+  const [warning, setWarning] = useState(false);
+  let sizes;
+  let availableQuantities;
+  let quantities;
 
-  for (var key in props.currentStyle.skus) {
-    if (!props.currentStyle.skus[key].quantity) {
-      continue;
+  //If the user switches the style, then reset the state
+  useEffect(() => {
+    setSelectedSize(0);
+    setSelectedQuantity(0);
+  }, [style]);
+
+  //Handle a click on the cart
+  let handleAddCartClick = (validation) => {
+    console.log(selectedSize);
+    if (selectedSize === 0 && validation === true) {
+      setWarning(true);
+    } else {
+      addToCart({
+        product_id: product.id,
+        style_id: style ? style.style_id : null,
+        sku_id: style ? Object.keys(style.skus)[selectedSize] : null,
```

Comment:
[code smell](https://en.wikipedia.org/wiki/Code_smell#:~:text=In%20computer%20programming%2C%20a%20code,%2C%20developer%2C%20and%20development%20methodology.&text=It%20is%20also%20a%20term%20used%20by%20agile%20programmers.) says this looks a little risky